### PR TITLE
Update To Use API Version 2019-10

### PR DESCRIPTION
**Pull request by forgiv**

### CODE REVIEW

File: src/Helpers/Endpoint.php [hunk 1]
```diff
@@ -36,11 +36,7 @@ public function __construct(Shopify $client)
      */
     public function __get($property)
     {
-        if (isset($this->$property)) {
-            return $this->$property;
-        }
-
-        return $this->client->__get($property);
+        return $this->$property ?? $this->client->__get($property);
```
Comment (danrichards) [inline src/Helpers/Endpoint.php:39]: What is the property is null?

Screen Shot 2020-04-13 at 9 15 01 AM

File: src/Models/AbstractModel.php [hunk 2]
```diff
@@ -70,9 +71,7 @@ public function __construct($data = [])
      */
     public function getKey()
     {
-        return isset($this->original[static::$identifier])
-            ? $this->original[static::$identifier]
-            : null;
+        return $this->original[static::$identifier] ?? null;
```
Comment (danrichards) [inline src/Models/AbstractModel.php:74]: It's ok here because it failsover to null

File: src/Shopify.php [hunk 3]
```diff
@@ -218,40 +246,90 @@ public static function make($shop, $token)
...
+        // If cursors haven't been set, then just call get normally.
+        if (empty($this->cursors)) {
+            return $this->get($query, $append);
         }
 
-        return $data;
+        // Only limit key is allowed to exist with cursor based navigation
+        foreach (array_keys($query) as $key) {
+            if ($key !== 'limit') {
+                Log::warning(__METHOD__.': Use of unallowed query param with cursored navigation');
```
Comment (danrichards) [inline src/Shopify.php]: Some people using this package will not be using Laravel.

File: src/Shopify.php [hunk 4]
```diff
@@ -580,7 +671,7 @@ private static function makeUri($api, $ids = [], $queue = [], $append = '', $bas
             $endpoint = implode('/', array_filter($parent)).'/'.$endpoint;
         }
 
-        $endpoint = '/'.$base.'/'.$endpoint;
+        $endpoint = '/'.$base.'/api/'.static::API_VERSION.'/'.$endpoint;
```
Comment (danrichards) [inline src/Shopify.php]: Can you switch this to a env variable?

```
$endpoint = defined('LARAVEL_START')
    ?  "/{$base}/api/".config('shopiify.api_version', static::API_VERSION)."/{$endpoint}"
    : "/{$base}/api/".static::API_VERSION."/{$endpoint}";
```

File: src/Shopify.php (hunk 3 above)
Comment (danrichards) [reply src/Shopify.php]: Maybe 

```
defined('LARAVEL_START')
    ? Log::warning();
    : // see `https://www.php.net/manual/en/function.error-log.php`
```

Also, it might be prettier to set a flag on the class for `$this->laravel` and add a `isLaravel()`

**Review by danrichards — CHANGES_REQUESTED** [review]
Few changes, thhank you

File: src/Shopify.php [hunk 5]
```diff
@@ -218,40 +246,90 @@ public static function make($shop, $token)
...
+            if ($key !== 'limit') {
+                Log::warning(__METHOD__.': Use of unallowed query param with cursored navigation');
+
+                return [];
+            }
+        }
+
+        // If cursors have been set and next hasn't been set, then return null.
+        if (empty($this->cursors['next'])) {
+            return null;
```
Comment (danrichards) [inline src/Shopify.php]: Return `[]`

**Review by danrichards — CHANGES_REQUESTED** [review]
Also, return `[]` here.

File: src/Helpers/Endpoint.php (hunk 1 above)
Comment (forgiv) [reply src/Helpers/Endpoint.php:39]: Looks like `isset` also returns false on a variable that's set to null.
Screen Shot 2020-04-13 at 5 41 27 PM

File: src/Shopify.php [hunk 6]
```diff
@@ -580,7 +677,9 @@ private static function makeUri($api, $ids = [], $queue = [], $append = '', $bas
             $endpoint = implode('/', array_filter($parent)).'/'.$endpoint;
         }
 
-        $endpoint = '/'.$base.'/'.$endpoint;
+        $endpoint = Util::isLaravel()
```
Comment (danrichards) [inline src/Shopify.php:680]: @shinyhiram Just ate about 90 minutes trying to figure out what happened here.

Luckily, I'm also using this project on a personal project on your latest version.

Requests were failing with weirdness.

TLDR. `config('shopify.api_base')` was intended to take the version (if any). So the new `config('shopify.api_version')` wreaked havoc.

In addition to that, my env (personal app) and production, already set the api_version in `config('shopify.api_base')` so I had to punch myself in the face until I figured out it was an old env var breaking shit.

Screen Shot 2020-04-25 at 9 13 02 PM

Screen Shot 2020-04-25 at 9 16 15 PM
